packages/flutter/lib/src/painting/matrix_utils.dart

```diff
@@ -163,7 +163,7 @@ class MatrixUtils {
   /// 0.0 before computing its bounding rect.
   static Rect inverseTransformRect(Matrix4 transform, Rect rect) {
     assert(rect != null);
-    assert(transform.determinant != 0.0);
```

Comment:
How did this pass our tests in debug mode? Is it returning true because a function is not equal to 0.0? Shouldn't it fail if we're trying to compare two things with different types?

Reply:
Well, a function tear-off is never equal to 0.0. In this case, the signature of the method is `bool operator.==(Object other)`, so this is a valid comparison.

Reply:
That seems to be a potential problem that could have a wide impact. Can we add a rule in `dart analyze` such that these kinds of comparisons would generate a compile time warning?

Reply:
hmm, we do have unrelated_type_equality_checks enabled. I'm not sure why this didn't get flagged though.

cc @pq

Reply:
Opened https://github.com/dart-lang/linter/issues/1525 to investigate. 👍

Reply:
Tricky. 0.0 cannot be equal to a tear-off, but for `class C { double call() => 7.0; }` and `C c = C();`, `c` could be equal to `transform.determinant` (both are zero-arg functions that return a double). Tricky.

Reply:
We should file a bug.